solr/solrj/src/java/org/apache/solr/client/solrj/io/stream/FacetStream.java

```diff
@@ -537,14 +555,29 @@ public void setStreamContext(StreamContext context) {
   }
 
   public void open() throws IOException {
-    if(cache != null) {
+    if (cache != null) {
       cloudSolrClient = cache.getCloudSolrClient(zkHost);
     } else {
       final List<String> hosts = new ArrayList<>();
       hosts.add(zkHost);
       cloudSolrClient = new Builder(hosts, Optional.empty()).withSocketTimeout(30000).withConnectionTimeout(15000).build();
     }
 
+    // Parallelize the facet expression across multiple collections for an alias using plist if possible
+    if (params.getBool("plist", defaultPlistEnabled)) {
```

Comment:
this is the main hook to see if the auto-plist approach should apply to this facet expression